README.md

```diff
@@ -207,31 +162,125 @@ Users want to determine several types of the formatting width as following
-            minutes: 46,
-            seconds: 40
-        });
-
-    console.log(formatter.format(duration));
-    // output: 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en').format(Temporal.Duration.from('PT2H46M40S')); // => 2 hr 46 min 40 sec
+    new Intl.DurationFormat('en', {
+      hours: 'numeric',
+      seconds: 'numeric',
+    }).format(Temporal.Duration.from('PT2H40S')); // => 2:00:40
+```
+
+### Constructor
+
+#### Syntax
+
+```javascript
+new Intl.DurationFormat(locales, options)
 ```
+
+#### Parameters
+
+* `locales` (`Array` | `String`): A locale string or a list of locale strings in decreasing order of preference.
+* `options` (optional `Object`): An object for configuring the behavior of the instance. It may have some or all of the following properties:
+  * `localeMatcher` (`String`): A string denoting which locale matching algorithm to use. Can be either `"best-fit"` or `"lookup"`. Defaults to `"best fit"`.
+  * `numberingSystem` (`String`): A string containing the name of the numbering system to be used for number formatting. Defaults to `undefined`.
+  * `style` (`String`): The base style to be used for formatting. This can be overriden per-unit by setting the more granular options. Can be one of `"long"`, `"short"` or `"narrow"`. Defaults to `"short"`.
+  * `years` (`String`): The style to be used for formatting years. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `yearsDisplay` (`String`): When to display years. `"auto"` means that years won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `years` is `undefined` and `"always"` otherwise.
+  * `months` (`String`): The style to be used for formatting months. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `monthsDisplay` (`String`): When to display months. `"auto"` means that months won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `months` is `undefined` and `"always"` otherwise.
+  * `weeks` (`String`): The style to be used for formatting weeks. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `weeksDisplay` (`String`): When to display weeks. `"auto"` means that weeks won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `weeks` is `undefined` and `"always"` otherwise.
+  * `days` (`String`): The style to be used for formatting days. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
+  * `daysDisplay` (`String`): When to display days. `"auto"` means that days won't be displayed if they are zero. Can be either `"always"` or `"auto"`. Defaults to `"auto"` if `days` is `undefined` and `"always"` otherwise.
+  * `hours` (`String`): The style to be used for formatting hours. Can be one of `"long"`, `"short"`, or `"narrow"`. Defaults to the value of `style`.
```

Comment:
Don't time units have different display options, e.g. including `'numeric'` and sometimes `'2-digit'`?

> Defaults to the value of `style`.

Also, if value of `style` is `'digital'` then this statement is not correct. I'd suggest saying "Its default depends on `style`, and then expanding the docs for `style` to explain the defaults, e.g. 
```
When `"long"`, `"short"`, or `"narrow"`, that same style will be used as the default for `'years'`, `'months'`, and all other unit options.
When `'digital'`, date units will default to `'narrow'` and time units to ????? (I wasn't sure if default was numeric or 2-digit).
```

Reply:
@justingrant I will explain the defaults in a different way. In the meantime, I added the missing options.